### PR TITLE
Sync NetworkController upsertNetworkConfiguration w/ core

### PR DESCRIPTION
**Pull request by mcmire**
## Explanation

This makes it easier to visually compare differences in the NetworkController unit tests between core and this repo.

Progresses https://github.com/MetaMask/core/issues/1197.

For reference, see the core repo: https://github.com/MetaMask/core/blob/e6326b7f2f50d6a2f6d72df1a66301f8c00d94ee/packages/network-controller/tests/NetworkController.test.ts#L2935

## Manual Testing Steps

There should be no functional changes. All tests should pass.

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by mcmire** [comment]
Blocked by https://github.com/MetaMask/core/pull/1413

File: app/scripts/controllers/network/network-controller.test.ts [hunk 1]
```diff
@@ -3919,7 +3997,7 @@ describe('NetworkController', () => {
             },
             {
               referrer: 'https://test-dapp.com',
-              source: MetaMetricsNetworkEventSource.Dapp,
+              source: 'dapp',
```
Comment (digiwand) [inline app/scripts/controllers/network/network-controller.test.ts:4000]: Q: do we intend to add the constant back in with the core tests after the sync?

**Review by digiwand — APPROVED** [review]

File: app/scripts/controllers/network/network-controller.test.ts (hunk 1 above)
Comment (mcmire) [reply app/scripts/controllers/network/network-controller.test.ts:4000]: Yes, I want to do a full audit of these tests at a later point once they are all in `core`, but I think that would be a good change specifically. It certainly is clearer and matches our conventions.

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Pull request merged**